### PR TITLE
Add file health check similar to folder health check

### PR DESCRIPTION
**Pull request by michaelmairegger**
This health check allows to check fo existing files similar to FolderHealthCheck

    services.AddHealthChecks()
        .AddFile(setup => setup.AddFile("FileUnderTest")), tags: new string[] { "file" });

### CODE REVIEW

File: src/HealthChecks.System/DependencyInjection/SystemHealthCheckBuilderExtensions.cs [hunk 1]
```diff
@@ -275,4 +276,62 @@ public static class SystemHealthCheckBuilderExtensions
             tags,
             timeout));
     }
+
+    /// <summary>
+    /// Add a healthcheck that allows to check for the existence of one or more files.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+    /// <param name="setup">Delegate for configuring the health check. Optional.</param>
+    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'files' will be used for the name.</param>
```
Comment (sungam3r) [inline src/HealthChecks.System/DependencyInjection/SystemHealthCheckBuilderExtensions.cs:285]:
```suggestion
    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'file' will be used for the name.</param>
```

File: src/HealthChecks.System/FileHealthCheckOptions.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+namespace HealthChecks.System;
+
+public class FileHealthCheckOptions
+{
+    internal List<string> Files { get; } = new();
```
Comment (sungam3r) [inline src/HealthChecks.System/FileHealthCheckOptions.cs:5]:
```suggestion
    public List<string> Files { get; } = new();
```

Why internal?

File: src/HealthChecks.System/FileHealthCheck.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        _fileOptions = fileOptions;
+    }
+
+    public Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            List<string> errorList = new();
+            foreach (string folder in _fileOptions.Files)
```
Comment (sungam3r) [inline src/HealthChecks.System/FileHealthCheck.cs:19]:
```suggestion
            foreach (string file in _fileOptions.Files)
```

File: src/HealthChecks.System/FileHealthCheck.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            {
+                if (!string.IsNullOrEmpty(folder))
```
Comment (sungam3r) [inline src/HealthChecks.System/FileHealthCheck.cs:21]:
```suggestion
                if (!string.IsNullOrEmpty(file))
```

File: src/HealthChecks.System/FileHealthCheck.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                {
+                    if (!File.Exists(folder))
```
Comment (sungam3r) [inline src/HealthChecks.System/FileHealthCheck.cs:23]:
```suggestion
                    if (!File.Exists(file))
```

File: src/HealthChecks.System/FileHealthCheck.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                    {
+                        errorList.Add($"File {folder} does not exist.");
```
Comment (sungam3r) [inline src/HealthChecks.System/FileHealthCheck.cs:25]:
```suggestion
                        errorList.Add($"File {file} does not exist.");
```

File: test/HealthChecks.System.Tests/HealthChecks.System.approved.txt [hunk 7]
```diff
@@ -14,6 +14,18 @@ namespace HealthChecks.System
...
     }
+    public class FileHealthCheck : Microsoft.Extensions.Diagnostics.HealthChecks.IHealthCheck
+    {
+        public FileHealthCheck(HealthChecks.System.FileHealthCheckOptions fileOptions) { }
+        public System.Threading.Tasks.Task<Microsoft.Extensions.Diagnostics.HealthChecks.HealthCheckResult> CheckHealthAsync(Microsoft.Extensions.Diagnostics.HealthChecks.HealthCheckContext context, System.Threading.CancellationToken cancellationToken = default) { }
+    }
+    public class FileHealthCheckOptions
+    {
+        public FileHealthCheckOptions() { }
+        public bool CheckAllFiles { get; set; }
```
Comment (sungam3r) [inline test/HealthChecks.System.Tests/HealthChecks.System.approved.txt:25]:
```suggestion
        public bool CheckAllFiles { get; set; }
        public System.Collections.Generic.List<string> Files { get; }
```

**Comment by sungam3r** [comment]
@michaelmairegger Thanks. I'm fine to merge after you add xml comments for new public APIs and some notes in README.

**Comment by michaelmairegger** [comment]
@sungam3r The xml comment is already existing. The new api for files is commented the same as the existing api for folders.

**Review by sungam3r — APPROVED** [review]

**Pull request merged**